controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1401,6 +1433,89 @@ int bml_internal::get_dcs_continuous_scan_params(const std::string &mac, int *ou
+                                       const unsigned int max_results_size,
+                                       uint8_t *output_result_status, bool is_single_scan)
+{
+    if (m_sockPlatform == nullptr && !connect_to_platform()) {
+        return (-BML_RET_CONNECT_FAIL);
+    }
+
+    // Initialize the promise for receiving the response
+    beerocks::promise<int> prmChannelScanResultsGet;
+    m_prmChannelScanResultsGet    = &prmChannelScanResultsGet;
+    int iOpTimeout                = DELAYED_RESPONSE_TIMEOUT; // Default timeout
+    auto scan_results             = std::list<beerocks_message::sChannelScanResults>();
+    uint32_t scan_results_size    = 0;
+    uint32_t scan_results_maxsize = uint32_t(max_results_size);
+
+    m_scan_results         = &scan_results;
+    m_scan_results_size    = &scan_results_size;
+    m_scan_results_maxsize = &scan_results_maxsize;
+    m_scan_results_status  = output_result_status;
+
+    //CMDU message
+    auto request = message_com::create_vs_message<
+        beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST>(cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR) << "Failed building cACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->radio_mac() = network_utils::mac_from_string(mac);
+    request->scan_mode() = (is_single_scan) ? 1 : 0;
+    // Build and send the message
+    if (!message_com::send_cmdu(m_sockMaster, cmdu_tx)) {
+        LOG(ERROR) << "Failed sending param get message!";
+        m_prmChannelScanResultsGet = nullptr;
+        m_scan_results             = nullptr;
+        m_scan_results_size        = nullptr;
+        m_scan_results_maxsize     = nullptr;
+        m_scan_results_status      = nullptr;
+        return (-BML_RET_OP_FAILED);
+    }
+    LOG(DEBUG) << "ACTION_BML_CHANNEL_SCAN_GET_RESULTS_REQUEST sent";
+
+    int iRet = BML_RET_OK;
+
+    if (!m_prmChannelScanResultsGet->wait_for(iOpTimeout)) {
+        LOG(WARNING) << "Timeout while waiting for results get response...";
+        iRet = -BML_RET_TIMEOUT;
+    }
+
+    // Clear the scan results members
+    m_scan_results         = nullptr;
+    m_scan_results_size    = nullptr;
+    m_scan_results_maxsize = nullptr;
+    m_scan_results_status  = nullptr;
+
+    // Clear the promise holder
+    m_prmChannelScanResultsGet = nullptr;
+
+    if (iRet != BML_RET_OK) {
+        LOG(ERROR) << "Results get failed!";
+        return (iRet);
+    }
+
+    iRet = prmChannelScanResultsGet.get_value();
+    if (iRet != int(eChannelScanOpErrCode::CHANNEL_SCAN_OP_SUCCESS)) {
+        LOG(ERROR) << "Results returned with error code:" << iRet << ". Aborting!";
+        return iRet;
+    }
+
+    //output_results_size will be set to the number of actual returning results
+    *output_results_size = 0;
+    for (auto res : scan_results) {
+        auto out = &((*output_results)[*output_results_size]);
```

Comment:
`auto &`

Reply:
Fixed

Reply:
I meant to pass the output result as a reference.

Change the declaration of `translate_channel_scan_results` to accept `BML_NEIGHBOR_AP &res_out` instead of `BML_NEIGHBOR_AP *res_out` and set `auto &out = (*output_results)[*output_results_size];`

Reply:
Oh.. my bad
Fixed and added a comment explaining the whole `(*output_results)[*output_results_size]` deal so it would be more understandable